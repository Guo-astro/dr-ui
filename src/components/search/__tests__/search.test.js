import React from 'react';
import Search from '../search';
import renderer from 'react-test-renderer';
import visit from 'unist-util-visit';

test('Search component, no options', () => {
  const component = renderer.create(<Search />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  visit(tree, 'input', node => {
    expect(node.props.placeholder).toBe('Search docs.mapbox.com');
  });
});

test('Search component, placeholder', () => {
  const component = renderer.create(<Search placeholder="Search me, ok?" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  visit(tree, 'input', node => {
    expect(node.props.placeholder).toBe('Search me, ok?');
  });
});

test('Search component, collapse', () => {
  const component = renderer.create(<Search collapse={true} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  visit(tree, 'input', node => {
    expect(node.props.placeholder).toBe('');
  });
});
