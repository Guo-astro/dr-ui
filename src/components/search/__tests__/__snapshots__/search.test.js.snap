// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component, collapse 1`] = `
<div
  className="h36 relative"
>
  <div
    className="w36-mm"
  >
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="docs-search-label"
      aria-owns={null}
      role="combobox"
    >
      <label
        className="cursor-pointer"
        htmlFor="docs-search-input"
        id="docs-search-label"
      >
        <div
          className="absolute flex-parent flex-parent--center-cross flex-parent--center-main w36 h36"
        >
          <svg
            className="icon color-gray"
          >
            <title>
              Search
            </title>
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </div>
      </label>
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-labelledby="docs-search-label"
        autoComplete="off"
        className="input bg-white px30 px0-mm color-transparent-mm"
        id="docs-search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Search component, no options 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="docs-search-label"
      aria-owns={null}
      role="combobox"
    >
      <label
        className="cursor-pointer"
        htmlFor="docs-search-input"
        id="docs-search-label"
      >
        <div
          className="absolute flex-parent flex-parent--center-cross flex-parent--center-main w36 h36"
        >
          <svg
            className="icon color-gray"
          >
            <title>
              Search
            </title>
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </div>
      </label>
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-labelledby="docs-search-label"
        autoComplete="off"
        className="input bg-white px30"
        id="docs-search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search docs.mapbox.com"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`Search component, placeholder 1`] = `
<div
  className="h36 relative"
>
  <div>
    <div
      aria-expanded={false}
      aria-haspopup="listbox"
      aria-labelledby="docs-search-label"
      aria-owns={null}
      role="combobox"
    >
      <label
        className="cursor-pointer"
        htmlFor="docs-search-input"
        id="docs-search-label"
      >
        <div
          className="absolute flex-parent flex-parent--center-cross flex-parent--center-main w36 h36"
        >
          <svg
            className="icon color-gray"
          >
            <title>
              Search
            </title>
            <use
              xlinkHref="#icon-search"
            />
          </svg>
        </div>
      </label>
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-labelledby="docs-search-label"
        autoComplete="off"
        className="input bg-white px30"
        id="docs-search-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search me, ok?"
        value=""
      />
    </div>
  </div>
</div>
`;
